rpc/core/blocks.go

```diff
@@ -180,9 +180,7 @@ func (env *Environment) BlockSearch(
 	pagePtr, perPagePtr *int,
 	orderBy string,
 ) (*ctypes.ResultBlockSearch, error) {
-
-	// skip if block indexing is disabled
-	if _, ok := env.BlockIndexer.(*blockidxnull.BlockerIndexer); ok {
+	if len(env.EventSinks) == 0 || env.EventSinks[0].Type() == indexer.NULL {
 		return nil, errors.New("block indexing is disabled")
 	}
```

Comment:
Let's create a function in the `indexer` package: `IndexingEnabled(sinks []EventSink) bool`.